Oct 3017 broadcaster with shell control to switch between bap presets

alexsven opened the pull request.

alexsven committed:
applications: nrf5340_audio: Use three level index per stream

- Each stream now has a three level index
 - Group
 - Subgroup
 - Stream_number
- Change le_audio_tx to accept new index and cap streams
- OCT-3042

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>

jayteemo committed:
net: lib: nrf_cloud_rest: add shadow transform request function

Add function nrf_cloud_rest_shadow_transform_request.
IRIS-9188

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

alexsven committed:
applications: audio: Create application for BCT

- Added all files needed to compile and flash
- Able to take preset and start broadcaster
- Able to restart and switch between presets
- Altered broadcast_send to take subgroup_index
- Will work with nrf5340_dk as well
- OCT-3017

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>

kapi-no committed:
samples: coremark: use minimal logging mode

Use the minimal logging mode in the CoreMark sample. Removed the flush
logic in the sample as the minimal logging mode works synchronously.

Ref: NCSDK-26794

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

kapi-no committed:
samples: bluetooth: fast_pair: locator_tag: improve ui doc

Improved the UI documentation in the Fast Pair Locator Tag sample
by moving the Thingy:53 related content to the tab view.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

kapi-no committed:
samples: bluetooth: fast_pair: locator_tag: improve config doc

Improved the configuration documentation in the Fast Pair Locator Tag
sample by moving the Thingy:53 related content to the tab view.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

jayteemo committed:
samples: cellular: nrf_cloud_rest_fota: add config support

Add support for setting the FOTA update check interval using
the config section in the shadow.
IRIS-9188

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

D-Triveni committed:
samples: wifi: Fix deprecated configuration variables

Replace deprecated OVERLAY_CONFIG and DTC_OVERLAY_FILE
with EXTRA_CONF_FILE and EXTRA_DTC_OVERLAY_FILE.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

alexsven commented on applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv.c:
```diff
@@ -368,6 +368,31 @@ int bt_mgmt_adv_buffer_put(struct bt_data *const adv_buf, uint32_t *index, size_
+		return ret;
+	}
+
+	ret = bt_le_per_adv_stop(ext_adv[ext_adv_index]);
+	if (ret) {
+		LOG_ERR("Failed to top periodic advertising: %d", ret);
+		return ret;
+	}
```
Check if per_adv is actually present (for unicast_servers)

alexsven replied:
Split into two functions, one for per_adv and one for ext_adv

andvib commented on applications/audio/broadcast_configuration_tool/CMakeLists.txt:
```diff
@@ -0,0 +1,29 @@
+
+# Include application events and configuration headers
+zephyr_library_include_directories(app PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/../../nrf5340_audio/include
```
```suggestion
    ${ZEPHYR_NRF_MODULE_DIR}/applications/nrf5340_audio/include
```
This way it doesn't break if the application is moved

andvib commented on applications/audio/broadcast_configuration_tool/CMakeLists.txt:
```diff
@@ -0,0 +1,29 @@
+)
+
+zephyr_library_include_directories(app PRIVATE
+    ${ZEPHYR_NRF_MODULE_DIR}/boards/arm/nrf5340_audio_dk_nrf5340)
```
Is this an old include? `nrf/boards/arm/nrf5340_audio_dk_nrf5340` doesn't exist anymore

alexsven replied:
Probably, will remove

andvib commented on applications/audio/broadcast_configuration_tool/CMakeLists.txt:
```diff
@@ -0,0 +1,29 @@
+target_sources_ifdef(CONFIG_NRF5340_AUDIO_SD_CARD_MODULE app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../nrf5340_audio/src/modules/sd_card.c)
+
+
+FILE(GLOB app_sources src/*.c)
```
This looks a bit scary - won't this include all .c files regardless of defines? E.g. the line above checking for SD card won't matter as the file is added here either way

alexsven replied:
As far as I know it will just include all the c-files in the broadcast_configuration_tool/src folder, which is just one main.c for now

andvib commented on applications/audio/broadcast_configuration_tool/Kconfig:
```diff
@@ -0,0 +1,26 @@
+
```
Copyright header missing

andvib commented on applications/audio/broadcast_configuration_tool/Kconfig.defaults:
If we're aiming to not have `Kconfig.defaults` files in the main application, should we avoid adding more defaults here? Might lead to more kconfigs in prj.conf until it's resolved in the main application, but that's easier to fix from a prj.conf

alexsven replied:
Can try to switch them all over to be in the prj.conf

alexsven replied:
Done

andvib commented on applications/audio/broadcast_configuration_tool/Kconfig.defaults:
```diff
@@ -0,0 +1,127 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
```
```suggestion
# Copyright (c) 2024 Nordic Semiconductor ASA
```

andvib commented on applications/audio/broadcast_configuration_tool/src/main.c:
```diff
@@ -0,0 +1,504 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
```
```suggestion
 * Copyright (c) 2024 Nordic Semiconductor ASA
```

andvib commented on applications/audio/broadcast_configuration_tool/src/main.c:
```diff
@@ -0,0 +1,504 @@
+	{.uuid_buf = &uuid_data2,
+	 .pba_metadata_vacant_cnt = BROADCAST_SOURCE_PBA_METADATA_VACANT,
+	 .pba_buf = pba_data[1]}};
+#else
```
Knowledge Sharing: Will we ever want to use this as not an Auracaster? What is the difference between a broadcaster that's an Auracaster and one that's not?

alexsven replied:
It depends on how the adv data is structured and it also has to support 16kHz. I guess SIG will be the most happy if everything is Auracasters

andvib commented on applications/audio/broadcast_configuration_tool/sysbuild/ipc_radio/prj.conf:
```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
```suggestion
# Copyright (c) 2024 Nordic Semiconductor ASA
```

andvib commented on applications/nrf5340_audio/src/modules/Kconfig:
```diff
@@ -178,6 +170,14 @@ config SD_CARD_PLAYBACK_THREAD_PRIORITY
 	int "Priority for the SD card playback thread"
 	default 7
 
+module = MODULE_SD_CARD
```
This will only be ran if `SD_CARD_PLAYBACK` is enabled, but the SD card module is included with `NRF5340_AUDIO_SD_CARD_MODULE`, so it should be moved out of the if. (Not sure if we have a use case for the SD card without the playback module tho)

rick1082 commented on applications/audio/broadcast_configuration_tool/src/main.c:
```diff
@@ -0,0 +1,504 @@
+/* Buffer for the UUIDs. */
+#define EXT_ADV_UUID_BUF_SIZE (128)
+NET_BUF_SIMPLE_DEFINE_STATIC(uuid_data, EXT_ADV_UUID_BUF_SIZE);
+NET_BUF_SIMPLE_DEFINE_STATIC(uuid_data2, EXT_ADV_UUID_BUF_SIZE);
```
Maybe we can add a description for explaining that the default setting is designed for two BIGs, so we have two UUID data sets.

alexsven replied:
This is just allocating the memory for it though, not actually advertising both sets. So I guess it will be part of a README as some point what the limits are

plskeggs committed:
samples: cellular: MSS: Thingy91X support

Add the Thingy91:x as an integration platform
and an allowed platform for both MQTT and CoAP.

Jira: IRIS-9259

Signed-off-by: Pete Skeggs <peter.skeggs@nordicsemi.no>

rick1082 commented on applications/audio/broadcast_configuration_tool/src/main.c:
```diff
@@ -0,0 +1,504 @@
+uint8_t pba_data[CONFIG_BT_ISO_MAX_BIG][BROADCAST_SRC_PBA_BUF_SIZE];
+
+static struct bt_bap_lc3_preset lc3_preset_8_2_1 = BT_BAP_LC3_BROADCAST_PRESET_8_2_1(
+	BT_AUDIO_LOCATION_FRONT_LEFT | BT_AUDIO_LOCATION_FRONT_RIGHT, BT_AUDIO_CONTEXT_TYPE_MEDIA);
```
Just curious, would there be a case that only mono channel output?

alexsven replied:
Yes, the location is not actually written here. This will be overwritten at a later stage, but the preset requires location to be set. I guess we could set it to 0 here and add a comment

rick1082 commented on applications/audio/broadcast_configuration_tool/src/main.c:
```diff
@@ -0,0 +1,504 @@
+
+	switch (msg->event) {
+	case BT_MGMT_EXT_ADV_WITH_PA_READY:
+		LOG_INF("Ext adv ready");
```
Would be great for EXT-ADV and PER-ADV ready.

alexsven replied:
Not sure what are you are asking for here

rick1082 replied:
Since the event is BT_MGMT_EXT_ADV_WITH_PA_READY, so it is not only "EXT ADV ready".

alexsven replied:
Ok, I see what you mean. The per adv is sort of in the ext adv, so in the context of this app there will never be an ext_adv without a per_avd, nor will there ever be a per_adv ready alone. We could probably also just remove this print all together

rick1082 commented on applications/audio/broadcast_configuration_tool/src/main.c:
```diff
@@ -0,0 +1,504 @@
+	bt_audio_codec_cfg_meta_set_bcast_audio_immediate_rend_flag(
+		&subgroups[0].group_lc3_preset.codec_cfg);
+
+	uint8_t english_src[3] = "eng";
```
Be careful, "eng" would be 4 elements including '\0'.
Probably we can replace that to uint8_t english_src[] = "eng"
So we won't trouble by the index issue.

alexsven replied:
sys_get_le24 requires a uint8_t[3] though. But I guess I can change to:
```suggestion
	uint8_t english_src[3] = {'e','n','g'};
```

rick1082 commented on applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.c:
```diff
@@ -209,6 +211,7 @@ static void bt_enabled_cb(int err)
 
 static int bonding_clear_check(void)
 {
+#if CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP
```
Since we are not going to provide link feature, probably we could remove this part ?

alexsven replied:
We need this here to be able to compile for nrf5340dk, this will also benefit OCT-2681

rick1082 commented on applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.h:
```diff
@@ -123,7 +123,14 @@ int bt_mgmt_scan_start(uint16_t scan_intvl, uint16_t scan_win, enum bt_mgmt_scan
 
 /**
- * @brief	Create and start advertising for ACL connection.
+ * @brief	Stop advertising for an ACL connection.
```
Might not related to this PR, but we could remove ACL, since ADV might not only for ACL connection, but just a BLE ADV like broadcasting.

andvib commented on applications/audio/broadcast_configuration_tool/prj.conf:
```diff
@@ -0,0 +1,140 @@
+CONFIG_TRANSPORT_BIS=y
+CONFIG_BT_AUDIO_BROADCAST_CONFIGURABLE=y
+
+CONFIG_SW_CODEC_LC3=n
```
nit: newline

andvib reviewed (approved):
Except from a missing newline it looks good!

olivier-le-sage committed:
doc: Add known issue about VS Set Connection Event Trigger command

Documents ppi channel bug for platforms using DPPI.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

wiba-nordic committed:
doc: Matter: clarify Bluetooth certification documentation

The Matter certification page now links to the Bluetooth certification
page. Also clarified the inheritance section so it can't be read
as being able to inherit Matter certification.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

rugeGerritsen committed:
doc: known_issues: Make DRGN-16013 apply to v2.6 and v2.7

The issue is also present in these releases.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

b-gent committed:
doc: bring back known issues to latest

Unorphan known issues after 2.7.0.

Signed-off-by: Bartosz Gentkowski <bartosz.gentkowski@nordicsemi.no>

nordic-piks committed:
tests: benchmarks: multicore: idle: add harness explicit

Common harness is appended not replaced by specific testcase.
harness_config is replaced by specific testcase.

Signed-off-by: Piotr Kosycarz <piotr.kosycarz@nordicsemi.no>

Commit:
samples: central_and_peripheral_hr: Fix coverity

This fixes coverity issues in Central and Peripheral
HR sample.

Jira: NCSDK-27145

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>
